source/common/network/socket_option_impl.h

```diff
@@ -41,7 +41,13 @@ typedef absl::optional<int> SocketOptionName;
 #define ENVOY_SOCKET_IPV6_FREEBIND Network::SocketOptionName()
 #endif
 
-class SocketOptionImpl : public Socket::Option, Logger::Loggable<Logger::Id::connection> {
+#ifdef TCP_FASTOPEN
+#define ENVOY_SOCKET_TCP_FASTOPEN Network::SocketOptionName(TCP_FASTOPEN)
+#else
+#define ENVOY_SOCKET_TCP_FASTOPEN Network::SocketOptionName()
```

Comment:
@htuch Do you mean just enforce TCP_FASTOPEN on all listeners by default and then run the tests?